### ISSUE TITLE
`ChatMemberUpdated.invite_link` missing when joining public groups

### ISSUE THREAD

**Issue by Bibo-Joshi** [tdlib/telegram-bot-api#428]
Hi. I did some testing and it looks like the `invite_link` field of the [`ChatMemberUpdated`](https://core.telegram.org/bots/api#chatmemberupdated) class is present only when joining private groups. For public groups, the field is missing even when the user does join via a link and not via the username of the group.
I expect that the added value of the invite link is smaller for public groups as they can be found by anyone, but this behavior is was unexpected as the documentation does not suggest it. If you could document this behavior or if the behavior could be updated, I'd be grateful.

**Comment by Poolitzer** [comment]
It works however when people join when pressing on a chat folder invite link.

**Comment by Bibo-Joshi** [comment]
Another insight: Apparently the link is there for groups that are newly created today …

**Comment by levlam** [comment]
> For public groups, the field is missing even when the user does join via a link and not via the username of the group.

This is not true. If the user joins the group via an invite link, the field `invite_link` is present. If the group is public, then users are free to join it directly, even the group was opened via an invite link. And this is exactly the behavior of most apps when user press "Join" button on the chat history screen.

**Comment by Bibo-Joshi** [comment]
Thanks for the quick reply! Your explanation about the "Join" button hiding away the `invite_link` does sound understandable. However, joining a newly created group via invite link + join button *does* give the `invite_link` attribute (just double checked again). So it seems to me that either the behavior is different for groups greated after some date or it is just inconsistent :/

**Comment by levlam** [comment]
The behavior depends on the used app. The official Android app is known to use direct join instead of invite link for public chats.

**Comment by Poolitzer** [comment]
I do not believe it is an app issue only.

I can use Telethon to join a chat with the same invite link with different accounts, in one chat member update it has the link, one doesn't

**Comment by levlam** [comment]
I tested this again multiple times and each time the `invite_link` field is present if the user joins the chat using the invite link. I've never managed to reproduce the issue if the chat is joined using the method `messages.importChatInvite`, and apps don't use the method to join public chats and join them directly, hence absence of the field `invite_link` is expected.
